src/Paysera/WalletApi/Mapper.php

```diff
@@ -2387,4 +2401,53 @@ public function encodeTransferConfiguration(Paysera_WalletApi_Entity_TransferCon
     {
         return (new Paysera_WalletApi_Mapper_TransferConfigurationMapper())->mapFromEntity($transferConfiguration);
     }
+
+    /**
+     * @param array $data
+     * @return Paysera_WalletApi_Entity_ClientPermissionsToWallet[]
+     */
+    public function decodeClientPermissionsToWallets(array $data)
+    {
+        $result = [];
+
+        foreach ($data as $clientPermissionsToWallet) {
+            $result[] = $this->decodeClientPermissionsToWallet($clientPermissionsToWallet);
+        }
+
+        return $result;
+    }
+
+    /**
+     * Decodes client wallet permissions
+     *
+     * @param array $data
+     *
+     * @return Paysera_WalletApi_Entity_ClientPermissionsToWallet
+     */
+    public function decodeClientPermissionsToWallet(array $data)
+    {
+        return Paysera_WalletApi_Entity_ClientPermissionsToWallet::create()
+            ->setWalletId($data['id'])
+            ->setAccountNumber($data['account'])
+            ->setScopes($data['scopes'])
+        ;
+    }
+
+    /**
+     * @param Paysera_WalletApi_Entity_ClientPermissionsToWallet $walletPermissions
+     * @return array
+     * @throws Paysera_WalletApi_Exception_LogicException
+     */
+    public function encodeClientPermissionsToWallet(Paysera_WalletApi_Entity_ClientPermissionsToWallet $walletPermissions)
+    {
+        if ($walletPermissions->getWalletId() === null) {
+            throw new Paysera_WalletApi_Exception_LogicException('Wallet ID must be provided');
+        }
+
+        return [
+            'id' => $walletPermissions->getWalletId(),
+            'account' => $walletPermissions->getAccountNumber(),
```

Comment:
you have a check with id, would it be fine if accountNumber or scopes not set?

Reply:
Added accountNumber checking